src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -209,15 +209,44 @@ function toggleCollapse(e, pans) {
     var expand = 'Expand';
     var collapse = 'Collapse';
     var panels = pans || $('div.panel-collapse');
+    var isElementTextStartsWithExpand = $(e).html().trim().startsWith(expand);
+
+    if (isElementTextStartsWithExpand) {
+        // Expand panels if the element text starts with expand.
+        // When a panel has class ajax_auto or ajax-response-auto, the panel data has not been loaded yet.
+        // We will load the data by Ajax, and ajax_auto or ajax-response-auto class will be removed.
+        // {@link #toggleCollapse} binds with clicking on both collapse panels button(button for all panels)
+        // and collapse panels button section (button for one panel). Class ajax_auto or ajax-response-auto
+        // is found in {@code panels} only if {@link #toggleCollapse} is invoked by clicking on collapse
+        // panels button.
+        var isCollapsePanelsButtonClicked = $(e).is($('#collapse-panels-button'));
 
-    if ($(e).html().trim().startsWith(expand)) {
         var i = 0;
         for (var idx = 0; idx < panels.length; idx++) {
+            if (isCollapsePanelsButtonClicked) {
+                // Might need to load Ajax if collapse panels button is clicked.
+                var $ajaxAuto = $(panels[idx]).parent().children('.ajax_auto');
+                var $ajaxResponseAuto = $(panels[idx]).parent().children('.ajax-response-auto');
+                var hasAjaxAutoLoading = $ajaxAuto.length !== 0;
+                var hasAjaxResponseAutoLoading = $ajaxResponseAuto.length !== 0;
+                if (hasAjaxAutoLoading) {
+                    $ajaxAuto.click();
+                }
+                if (hasAjaxResponseAutoLoading) {
+                    $ajaxResponseAuto.click();
+                }
+                if (hasAjaxAutoLoading || hasAjaxResponseAutoLoading) {
+                    // Ajax loading is triggered by clicking, therefore we do not need to expand the panel.
+                    continue;
+                }
+            }
+            // Expand the panel if the panel is collapsed at the moment
             if ($(panels[idx]).attr('class').indexOf('in') === -1) {
                 setTimeout(showSingleCollapse, 50 * i, panels[idx]);
                 i++;
             }
         }
+
```

Comment:
You didn't write this function, so it is not your fault if it is not very easy to understand in the first place. However, you are adding a huge branch to it, so the understandability is getting even worse now. Let's start from the top: What's the high-level algorithm/logic of the `toggleCollapse(e, pans)` function? My naiive expectations is that it is something like this:
```
if already expanded, 
    collapse
else 
    expand
``` 
But I can't seem to find such a high-level logic in the implementation.